controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -93,46 +89,56 @@ public void refresh() {
     }
 
     @Override
-    CompletableFuture<Void> checkStreamExists(Create create) {
-        CompletableFuture<Void> result = new CompletableFuture<>();
+    CompletableFuture<CreateStreamResponse> checkStreamExists(StreamConfiguration configuration, long timestamp) {
+        CompletableFuture<CreateStreamResponse> result = new CompletableFuture<>();
 
         long time = creationTime.get();
-        if (time != Long.MIN_VALUE && create.getCreationTime() != time) {
-            result.completeExceptionally(StoreException.create(StoreException.Type.DATA_EXISTS, getName()));
-        } else {
-            result.complete(null);
+        synchronized (lock) {
```

Comment:
the logic is as follows:
1. if the stream was not created earlier, --> return New
2. if stream was created earlier, if prev-creation-time == new-creation-time --> return New
3. if prev-creation-time != new-creation-time --> 
3.a if prev-configuration == null (if previous failure happened before configuration was written) then return New
3.b else if (state <= CREATING) return Exists_Creating else return Exists_Active


All this is blocked in synchronized block because we do reads of configuration and state under locks. 
hence we acquire lock in this method too for reading configuration and state